src/Interpreters/Cache/FileCache.cpp

```diff
@@ -247,7 +247,15 @@ CachePriorityGuard::Lock FileCache::lockCache() const
 
 CachePriorityGuard::Lock FileCache::tryLockCache(std::optional<std::chrono::milliseconds> acquire_timeout) const
 {
-    return acquire_timeout.has_value() ? cache_guard.tryLockFor(acquire_timeout.value()) : cache_guard.tryLock();
+    if (acquire_timeout.has_value())
+    {
+        ProfileEventTimeIncrement<Microseconds> watch(ProfileEvents::FilesystemCacheLockCacheMicroseconds);
```

Comment:
afaiu we'll wait in any case, so why not to account this time regardless of `acquire_timeout`?

Reply:
I thought tryLock without any timeout is quite quick so does not make sense to add to metric?